tests/test_move_and_copy.py

```diff
@@ -352,3 +352,83 @@ def test_copy_segment_abort(self, accesskey, url, tmp_path):
             dataset_client.copy_segment("Segment1", "Segment2")
 
         gas_client.delete_dataset(dataset_name)
+
+    def test_copy_segment_between_datasets(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name_1 = get_dataset_name()
+        dataset_name_2 = dataset_name_1 + "_2"
+        gas_client.create_dataset(dataset_name_1)
+        dataset_client_2 = gas_client.create_dataset(dataset_name_2)
+        dataset_1 = Dataset(name=dataset_name_1)
```

Comment:
I think you should separate the creation of Dataset1 and 2 to make the code more readable.